Program.cs

```diff
@@ -58,11 +60,58 @@ private static void ExtractLicense(Product product)
                     if (match.Success)
                     {
                         Console.WriteLine($"Found key for {product.Name}: {match.Captures[0]}");
+                        Console.WriteLine("\nDo you want to save it? (Yes/No)");
```

Comment:
This is not how console applications typically work. I suggest you take a look at some other common command line tools for inspiration. For instance, if you are using windows, try opening a command prompt session and typing

dir /?

You will see that all of the command line options are optional, but when specifying them you modify the behavior of the program. That means there is no need for an interactive user to answer yes/no or interactively giving additional options after the program has been executed, all the information is already there in the command line. This makes the command line tools very powerful, as they can then be combined in scripts, feeding data from one tool into the next.

If you want to have this change merged, I would like you to rewrite this part to only take input from the command line, not interactively with Console.ReadLine(). If you REALLY want an interactive mode, then at least have the user specify that explicitly, for instance using -interactive as a command line option.

I suggest you have a look at a library such as [commandline](https://github.com/commandlineparser/commandline) to help you avoid writing a lot of boilerplate code.